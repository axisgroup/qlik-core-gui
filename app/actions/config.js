// @flow
export const SET_CONFIG = 'SET_CONFIG';
export const REMOVE_CONFIG = 'REMOVE_CONFIG';

import type { Config } from '../reducers/types';

export function setConfig(config: Config) {
  return {
    type: SET_CONFIG,
    config
  };
}

export function removeConfig() {
  return {
    type: REMOVE_CONFIG,
    config: {}
  };
}
