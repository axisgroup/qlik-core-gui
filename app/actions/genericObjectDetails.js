// @flow
export const SET_TAB = 'SET_TAB';
export const SET_PROPERTIES = 'SET_PROPERTIES';

export function setTab(newTab: string) {
  return {
    type: SET_TAB,
    payload: typeof newTab === 'string' ? newTab : null
  };
}
