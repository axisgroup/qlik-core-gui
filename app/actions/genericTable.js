// @flow
export const TOGGLE_ROW = 'TOGGLE_ROW';
export const SELECT_OBJ = 'SELECT_OBJ';
export const TOGGLE_EXPAND_ALL = 'TOGGLE_EXPAND_ALL';
export const SAVE_SEARCH_TERM = 'SAVE_SEARCH_TERM';

export function toggleRow(rowName: string) {
  return {
    type: TOGGLE_ROW,
    payload: typeof rowName === 'string' ? rowName : null
  };
}

export function selectObj(obj: string) {
  return {
    type: SELECT_OBJ,
    payload: typeof obj === 'string' ? obj : null
  };
}

export function toggleExpandAll(expand?: boolean) {
  return {
    type: TOGGLE_EXPAND_ALL,
    payload: typeof expand === 'boolean' ? expand : null
  };
}

export function saveSearchTerm(term: string) {
  return {
    type: SAVE_SEARCH_TERM,
    payload: term
  };
}
