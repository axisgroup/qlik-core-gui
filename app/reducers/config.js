// @flow
import { INCREMENT_COUNTER, DECREMENT_COUNTER } from '../actions/counter';
import { SET_CONFIG, REMOVE_CONFIG } from '../actions/config';
import type { Action, Config } from './types';

export default function config(state: Config = {}, action: Action) {
  switch (action.type) {
    case SET_CONFIG:
      return action.payload || {};
    case REMOVE_CONFIG:
      return {};
    default:
      return state;
  }
}
