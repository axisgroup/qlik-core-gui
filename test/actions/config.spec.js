import { spy } from 'sinon';
import * as actions from '../../app/actions/config';

describe('actions', () => {
  it('should create a set config action', () => {
    expect(actions.setConfig({host: 'localhost', port: 9076, appname: 'drugcases.qvf'})).toMatchSnapshot();
  });

  it('should create a remove config action', () => {
    expect(actions.removeConfig()).toMatchSnapshot(); 
  });

  // it('should decrement should create decrement action', () => {
  //   expect(actions.decrement()).toMatchSnapshot();
  // });

  // it('should incrementIfOdd should create increment action', () => {
  //   const fn = actions.incrementIfOdd();
  //   expect(fn).toBeInstanceOf(Function);
  //   const dispatch = spy();
  //   const getState = () => ({ counter: 1 });
  //   fn(dispatch, getState);
  //   expect(dispatch.calledWith({ type: actions.INCREMENT_COUNTER })).toBe(true);
  // });

  // it('should incrementIfOdd shouldnt create increment action if counter is even', () => {
  //   const fn = actions.incrementIfOdd();
  //   const dispatch = spy();
  //   const getState = () => ({ counter: 2 });
  //   fn(dispatch, getState);
  //   expect(dispatch.called).toBe(false);
  // });

  // // There's no nice way to test this at the moment...
  // it('should incrementAsync', done => {
  //   const fn = actions.incrementAsync(1);
  //   expect(fn).toBeInstanceOf(Function);
  //   const dispatch = spy();
  //   fn(dispatch);
  //   setTimeout(() => {
  //     expect(dispatch.calledWith({ type: actions.INCREMENT_COUNTER })).toBe(
  //       true
  //     );
  //     done();
  //   }, 5);
  // });
});
