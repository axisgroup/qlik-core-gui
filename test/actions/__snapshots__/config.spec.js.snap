// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions should create a remove config action 1`] = `
Object {
  "config": Object {},
  "type": "REMOVE_CONFIG",
}
`;

exports[`actions should create a set config action 1`] = `
Object {
  "config": Object {
    "appname": "drugcases.qvf",
    "host": "localhost",
    "port": 9076,
  },
  "type": "SET_CONFIG",
}
`;
